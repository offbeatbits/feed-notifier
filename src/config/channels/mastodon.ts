import { boolean } from "../../utils";
import { MastodonSettings } from "../../types.d";

const { env } = process;

export const mastodon: MastodonSettings = {
  enabled: boolean(env.FN_MASTODON_ENABLE),
  username: env.FN_MASTODON_USERNAME!,
  host: env.FN_MASTODON_HOST!,
};
