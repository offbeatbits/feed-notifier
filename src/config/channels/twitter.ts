import { TwitterSettings } from '../../types';

const { env } = process;

export const twitter: TwitterSettings = {
  username: env.TWITTER_USERNAME!,
};
