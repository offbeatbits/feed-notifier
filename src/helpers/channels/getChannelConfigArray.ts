import {
  ChannelConfig,
  ChannelName,
  OAuth2ServiceSettings,
  ChannelSettings,
  ChannelConfigArray,
} from '../../types';

export const getChannelConfigArray = (
  channels: ChannelConfig<ChannelSettings>,
  clients: ChannelConfig<OAuth2ServiceSettings>,
) => (Object.keys(channels) as Array<ChannelName>).map(channelKey => ({
  type: channelKey,
  settings: {
    ...channels[channelKey],
    client: clients[channelKey],
  },
})) as ChannelConfigArray;
