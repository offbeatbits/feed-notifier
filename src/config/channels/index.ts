import { mastodon } from './mastodon';
import { twitter } from './twitter';

export const channels = {
  mastodon,
  twitter,
};
