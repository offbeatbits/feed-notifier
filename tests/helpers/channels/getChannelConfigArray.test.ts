import { getChannelConfigArray } from '../../../src/helpers/channels/getChannelConfigArray';

import {
  ChannelConfig,
  ChannelSettings,
  OAuth2ServiceSettings,
} from '../../../src/types';

const sampleChannels = {
  sampleChannel1: {
    foo: 'bar1',
    bar: 'baz1',
  },
  sampleChannel2: {
    foo: 'bar2',
    bar: 'baz2',
  },
} as unknown as ChannelConfig<ChannelSettings>;

const sampleClients = {
  sampleChannel1: {
    foo: 'bar1',
    bar: 'baz1',
  },
  sampleChannel2: {
    foo: 'bar2',
    bar: 'baz2',
  },
} as unknown as ChannelConfig<OAuth2ServiceSettings>;

describe('getChannelConfigArray', () => {
  it('should return channel config array', () => {
    expect(getChannelConfigArray(sampleChannels, sampleClients)).toMatchSnapshot();
  });
});
