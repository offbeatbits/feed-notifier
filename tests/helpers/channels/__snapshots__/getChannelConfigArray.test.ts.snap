// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getChannelConfigArray should return channel config array 1`] = `
Array [
  Object {
    "settings": Object {
      "bar": "baz1",
      "client": Object {
        "bar": "baz1",
        "foo": "bar1",
      },
      "foo": "bar1",
    },
    "type": "sampleChannel1",
  },
  Object {
    "settings": Object {
      "bar": "baz2",
      "client": Object {
        "bar": "baz2",
        "foo": "bar2",
      },
      "foo": "bar2",
    },
    "type": "sampleChannel2",
  },
]
`;
